notes/2024-03-26_architecture-meeting.md

```diff
@@ -0,0 +1,99 @@
+              “Hello world” on test cluster?
+                * TODO: Matthew Fisher Trey Stafford give rushirajnenuji@gmail.com our IP
+                  addresses for allowlist.
+                * TODO: rushirajnenuji@gmail.com to check if he has access to `k8s-dev`
+                * TODO: create new qgnet service account and namespace, see
+                  https://github.com/DataONEorg/k8s-cluster/tree/main/authorization
+                * There is some documentation on GH that can help getting started
+                  (<https://github.com/dataoneorg/k8s-cluster>)
+                * Next step: Group programming session to get connected and do “Hello
+                  world” on dev cluster.
+    * ADC getting better at Helm, which has improved DX over straight K8s YAML. Matt
+      Brooke is the go-to person for Helm Charts.
+    * Also working on a project with Google, and they have $alot in cloud credits.
+      Exploring transitioning to GCP-friendly kubernetes workflows. Want things to work
+      in both on-prem cluster and GCP.
+
+
+### Pangeo Forge collab/overlap
+
+* Aimee Barciauskas with DevSeed / Pangeo Forge is interested in our project and how it
+  aligns with Pangeo Forge goals to serve NASA needs.
+    * Now is the time to discuss with Aimee now our technical alignment and paths
+      forward collaborating with PF.
+* What is everyone thinking/feeling about the technical aspects of this opportunity
+  right now? Excited? Concerned? No opinion yet (that’s OK, we’re just starting to learn
+  🙂) We can talk about the “people” part of this opportunity in other meetings, e.g.
+  with Aimee B.
+    * (Matt J) Current workflows tend to be one-off. Want to move towards a system where
+      contributing a new dataset kicks off tooling that helps build needed visualization
+      data
+
+
+### Another potential stong partner: Source Cooperative
+
+* Matt J. working with Carl Boettiger
+    * https://source.coop/
+    * Matt F. works on Openscapes with Carl
+    * TODO: Matthew Fisher ask Carl and learn more!
+
+
+### Data-chunking communities of practice
+
+* (Matt J) There are two major communities of practice for data-chunking, and Pangeo
+  Forge falls strongly on one side of that (Zarr, Xarray, Dask, ...); the other side is
+  web 3d tiles, and this is where ADC has been focusing more. These approaches come from
+  two different use cases (data processing / hardware-accelerated web raster
+  visualization). ML models researchers are producing on PDG are vector outputs, and so
+  ADC has been focused on how to do fast 3d visualization of this type of data. E.g.
+  hundreds of kilometers large point-clouds displaying in the browser.
+
+
+### High-level arch. diagram
+
+<https://qgreenland-net.github.io/diagrams/high-level-architecture.html>
+
+How did we do? What other levels of diagram would be useful at this point? We like using
+the C4 diagramming framework to diagram various levels of a system.
```

Comment:
@mbjones if you have a free moment, can you comment on this? We left this for last because we felt synchronous discussion wasn't as important for this as the other topics we did discuss.

Reply:
The diagrams are great -- I really like Mermaid (and PlantUML before it) for dev diagrams because the source lives in the dev repos, and is easily displayed in markdown docs and on github repos and issues. Is that all you are asking about? I'll have to spend some more time thinking about the actual architecture before I have thougths on that.

Reply:
We were actually asking about the actual architecture :)

Rushiraj, Trey, and I discussed this and we're all big fans of Mermaid (diagrams-as-code in general) for the same reasons!

One thing that's frustrating is GitHub doesn't render QMD as Markdown, so the mermaid diagrams don't render. Quarto does require the QMD extension to render Mermaid. I was hoping there was some way we could tell Linguist to render QMD as Markdown, but I've been struggling to find it. If you know anything about this it'd be great to hear :)

Reply:
@mbjones we're going to merge this and keep rolling forward, could you put any architecture diagram feedback in a new issue?